File: LondonMeetup/app/src/main/java/com/suyashsrijan/londonmeetup/API/tfl/networking/HttpGetRequest.java
package com.suyashsrijan.londonmeetup.API.tfl.networking;

import android.os.AsyncTask;

import com.suyashsrijan.londonmeetup.API.tfl.interfaces.HttpGetResultCallback;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

public class HttpGetRequest extends AsyncTask<String, Void, String> {

    public static final String REQUEST_METHOD = "GET";
    public static final int READ_TIMEOUT = 15000;
    public static final int CONNECTION_TIMEOUT = 15000;

    private HttpGetResultCallback delegate = null;
    private Exception exception = null;

    public HttpGetRequest(HttpGetResultCallback callback) {
        this.delegate = callback;
    }

    @Override
    protected String doInBackground(String... params) {
        String urlString = params[0];
        String inputLine;
        String result = "";
        try {
            URL myUrl = new URL(urlString);
            HttpURLConnection connection = (HttpURLConnection) myUrl.openConnection();
            connection.setRequestMethod(REQUEST_METHOD);
            connection.setReadTimeout(READ_TIMEOUT);
            connection.setConnectTimeout(CONNECTION_TIMEOUT);
            connection.connect();

            InputStreamReader streamReader = new InputStreamReader(connection.getInputStream());
            BufferedReader reader = new BufferedReader(streamReader);
            StringBuilder stringBuilder = new StringBuilder();
            while((inputLine = reader.readLine()) != null){
                stringBuilder.append(inputLine);
            }
            reader.close();
            streamReader.close();
            result = stringBuilder.toString();
            return result;
        } catch (Exception e) {
            exception = e;
        }
        return result;
    }
    @Override
    protected void onPostExecute(String result){
        if (exception != null) {
            delegate.error(exception);
        } else{
            delegate.response(result);
        }
    }
}

File: LondonMeetup_v2/app/src/main/java/com/suyashsrijan/londonmeetup/oldcode/API/tfl/networking/HttpGetRequest.kt
package com.suyashsrijan.londonmeetup.API.tfl.networking

import android.os.AsyncTask

import com.suyashsrijan.londonmeetup.API.tfl.interfaces.HttpGetResultCallback

import java.io.BufferedReader
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.URL

class HttpGetRequest(callback: HttpGetResultCallback) : AsyncTask<String, Void, String>() {

    private var delegate: HttpGetResultCallback? = null
    private var exception: Exception? = null

    init {
        this.delegate = callback
    }

    override fun doInBackground(vararg params: String): String {
        val urlString = params[0]
        var inputLine: String
        var result = ""
        try {
            val myUrl = URL(urlString)
            val connection = myUrl.openConnection() as HttpURLConnection
            connection.requestMethod = REQUEST_METHOD
            connection.readTimeout = READ_TIMEOUT
            connection.connectTimeout = CONNECTION_TIMEOUT
            connection.connect()

            val streamReader = InputStreamReader(connection.inputStream)
            val reader = BufferedReader(streamReader)
            val stringBuilder = StringBuilder()
            reader.lines().forEach { line ->
                stringBuilder.append(line)
            }
            /*while ((inputLine = reader.readLine()) != null) {
                stringBuilder.append(inputLine)
            }*/
            reader.close()
            streamReader.close()
            result = stringBuilder.toString()
            return result
        } catch (e: Exception) {
            exception = e
        }

        return result
    }

    override fun onPostExecute(result: String) {
        if (exception != null) {
            delegate!!.error(exception!!)
        } else {
            delegate!!.response(result)
        }
    }

    companion object {

        val REQUEST_METHOD = "GET"
        val READ_TIMEOUT = 15000
        val CONNECTION_TIMEOUT = 15000
    }
}